### PR TITLE
Segmentation: Prediction

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

This implements the prediction phase for the DeepLab-based segmentation code.

### Checklist

- [x] Ran scripts/format_code and commited any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness
- [x] Test prediction packages
- [x] Unit tests
- [x] Requires https://github.com/azavea/raster-vision/pull/369
- [x] Requires https://github.com/azavea/raster-vision/pull/371

### CODE REVIEW

**Comment by jamesmcclain** [comment]
A scaled-down visualization of predictions on the first 128 chips.  The predictions themselves need work, but the visualization is confirmed correct.  If the prediction process had not been truncated after the 128th chip, this would have produced a prediction raster for the whole scene.

**Comment by jamesmcclain** [comment]
It is a bit difficult to demonstrate with a static picture, but these predictions look fairly good.

File: src/rastervision/protos/make_training_chips.proto [hunk 1]
```diff
@@ -57,6 +57,8 @@ message MakeTrainingChipsConfig {
         optional float debug_chip_probability = 1 [default=0.25];
         optional float empty_survival_probability = 2 [default=0.25];
         optional int32 number_of_chips = 3 [default=100];
+        optional int32 super_window_factor = 4 [default = 6];
+        optional int32 minor_axis_backoff = 5 [default = 0];
```
[the remark this answers is not in the document]
Comment (jamesmcclain) [reply src/rastervision/protos/make_training_chips.proto]: Done

File: src/rastervision/label_stores/segmentation_raster_file.py [hunk 2]
```diff
@@ -19,146 +24,260 @@ class SegmentationRasterFile(LabelStore):
...
-            (self.channels, _, _) = self.src._get_chip(small_box).shape
-        else:
-            self.channels = 1
-
-    def has_labels(self, window: Box) -> bool:
-        """Given a window, deterine whether there are any labels in it.
+    def interesting_subwindow(self, window: Box, size: int,
+                              shift: int) -> Union[Box, None]:
+        """Given a larger window, return a sub-window that contains interesting
+        pixels (pixels of class 1).
```
[the remark this answers is not in the document]
Comment (jamesmcclain) [reply src/rastervision/label_stores/segmentation_raster_file.py]: Done

**Comment by jamesmcclain** [comment]
Comments addressed.

File: src/rastervision/label_stores/segmentation_raster_file.py [hunk 3]
```diff
@@ -116,59 +116,41 @@ def set_labels(
...
-            elif argmax == 0 and not translated[0, 0] == 1:
-                return -1
-            elif size == larger_size:
-                return window
+
+            target_count = 0
+            for i in target_classes:
+                target_count = target_count + (translated == i).sum()
+
+            if target_count < (size * math.sqrt(size)):
```
Comment (lewfish) [inline src/rastervision/label_stores/segmentation_raster_file.py]: I'm not sure that `size * sqrt(size)` counts as "enough". If the only target class is cars for example, the `target_count` could be less than this. Maybe there should be an option for this, and it should be a threshold on the proportion of target pixels, similar to `ioa_thresh`?

File: src/rastervision/label_stores/segmentation_raster_file.py [hunk 4]
```diff
@@ -116,59 +116,41 @@ def set_labels(
             raise ValueError('Unsure how to handle source={}'.format(
                 type(source)))
 
-    def interesting_subwindow(self, window: Box, size: int,
-                              shift: int) -> Union[Box, None]:
-        """Given a larger window, return a sub-window that contains interesting
-        pixels (pixels of class 1).
+    def window_predicate(self, window: Box, target_classes: List[int]) -> bool:
```
Comment (lewfish) [inline src/rastervision/label_stores/segmentation_raster_file.py]: Consider renaming to `enough_target_pixels`

**Comment by jamesmcclain** [comment]
Comments addressed

**Pull request merged**